examples/xml-to-record-with-projection/xml_to_record_with_projection.bal

```diff
@@ -0,0 +1,34 @@
+import ballerina/data.xmldata;
+import ballerina/io;
+
+// Define a closed record type to capture the required elements and attributes from the XML data.
+type Book record {|
+    string name;
+    string author;
+|};
+
+public function main() returns error? {
+    xml xmlData = xml `
+        <book>
+            <name>Clean Code</name>
+            <author>Robert C. Martin</author>
+            <year>2008</year>
+            <publisher>Prentice Hall</publisher>
+        </book>`;
+
+    // Based on the expected type, it selectively converts the XML data to the record type.
+    Book book = check xmldata:parseAsType(xmlData);
+    io:println(book);
+
+    string xmlStr = string `
```

Comment:
imo, define this in module level is more clear. wdyt?

Reply:
Updated https://github.com/ballerina-platform/ballerina-distribution/pull/5295/commits/07acb43a5a2aa3ce5a5e0c72d5efe81e0ff0cd96